### PR TITLE
fix: update role labels of unattached PVCs

### LINKED ISSUE #1422: PVC role label not being updated if instance are 0
How to reproduce:

* Create cluster with 1 instance
* Scale down to 0 the operator
* Remove labels from PVC
* Scale operator to 1

After this the operator will not be able to find the PVC and will not start up the pod

### PR DESCRIPTION
**Pull request by mnencia**
Closes #1422 

Signed-off-by: Marco Nenciarini <marco.nenciarini@enterprisedb.com>

### CODE REVIEW

**Review by jsilvela — APPROVED** [review]

File: pkg/reconciler/persistentvolumeclaim/reconciler_test.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+	"github.com/cloudnative-pg/cloudnative-pg/pkg/specs"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	apiv1 "github.com/cloudnative-pg/cloudnative-pg/api/v1"
+	utils2 "github.com/cloudnative-pg/cloudnative-pg/pkg/utils"
```
Comment (mnencia) [inline pkg/reconciler/persistentvolumeclaim/reconciler_test.go:31]: Please fix the import alias

File: pkg/reconciler/persistentvolumeclaim/resources_test.go [hunk 2]
```diff
@@ -24,6 +24,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	apiv1 "github.com/cloudnative-pg/cloudnative-pg/api/v1"
+	utils2 "github.com/cloudnative-pg/cloudnative-pg/pkg/utils"
```
Comment (mnencia) [inline pkg/reconciler/persistentvolumeclaim/resources_test.go:27]: Please fix the `utils2` import alias also here

**Review by litaocdl — APPROVED** [review]

**Comment by litaocdl** [comment]
Manually verified with the patch (follow Jonathan’s steps in cnpg)

- create a cluster with 1 instance. 
- 
- scale the cnpg operator to 0.
- 
- delete the pvcRole in instance pvc
- 
- delete the instance pod
- 
- scale the cnpg operator to 1

with patch: the pvcRole label is added to instance pvc, and instance pod is started successfully.

without patch: the pvcRole label is not added to instance pvc, and no instance pod.

**Review by sxd — APPROVED** [review]

**Pull request merged**